backend/dataall/base/utils/logs_utils.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional, Callable, List, Any
+from dataall.base.config import config
+from dataall.base.utils.decorator_utls import process_func
+
+
+def is_feature_has_allowed_values(
+    allowed_values: List[Any], resolve_property: Optional[Callable] = None, config_property: Optional[str] = None
+) -> object:
+    def decorator(f):
+        fn, fn_decorator = process_func(f)
+
+        def decorated(*args, **kwargs):
+            config_property_value = None
+            if config_property is None and resolve_property is None:
+                raise Exception('Config property not provided')
+            if resolve_property:
+                config_property_value = resolve_property(*args, **kwargs)
+            value = config.get_property(config_property_value)
+            if value not in allowed_values:
+                raise Exception(
+                    f'Disabled since incorrect values in config {config_property_value}. Correct config values {allowed_values}'
+                )
+            return fn(*args, **kwargs)
+
+        return fn_decorator(decorated)
+
+    return decorator
+
+
+def check_if_user_allowed_view_logs(groups, config):
+    if (config == 'admin-only' and 'DAAdministrators' not in groups) or config == 'disabled':
+        return False
+    return True
```

Comment:
Unless we make this logic very generic and turn it into a decorator that disables service calls depending of if user is admin (out of scope for this PR) this code should be implemented in the stack module.

Reply:
In the same way there is a `ShareLogsService.check_view_log_permissions`

Reply:
Will do.  Since the code was getting repeated , here`StackService.check_if_user_allowed_view_logs` and here `ShareLogsService.check_view_log_permissions` I thought of putting it here.